Xamarin.PropertyEditing.Mac/Themes/MacThemeManager.cs

```diff
@@ -27,5 +29,23 @@ protected override void SetTheme ()
 					break;
 			}
 		}
+
+		public string GetImageNameForTheme (string imageNamed, string selectedSuffix = "")
+		{
+			return (Theme == PropertyEditorTheme.Dark ? imageNamed + "~dark" : imageNamed) + selectedSuffix;
+		}
+
+		public NSImage GetImageForTheme (string imageNamed, string selectedSuffix = "")
+		{
+			var imageNameForTheme = GetImageNameForTheme (imageNamed, selectedSuffix);
+			NSImage themeImage;
+			if (this.themeCache.TryGetValue (imageNameForTheme, out themeImage)) {
```

Comment:
@ermau I was contemplating wrapping this into something more async friendly. Thoughts?

Reply:
I don't think we want to complicate every image load with asyncifying it, especially since we'll only have the delay once.

Reply:
Sure.